File: domain/src/main/java/org/openmedstack/eventstore/IAccessSnapshots.java
package org.openmedstack.eventstore;

import java.util.concurrent.CompletableFuture;

public interface IAccessSnapshots {
    CompletableFuture<Snapshot> GetSnapshot(String bucketId, String streamId, int maxRevision);

    /// <summary>
    ///     Adds the snapshot provided to the stream indicated.
    /// </summary>
    /// <param name="snapshot">The snapshot to save.</param>
    /// <returns>If the snapshot was added, returns true; otherwise false.</returns>
    /// <exception cref="StorageException" />
    /// <exception cref="StorageUnavailableException" />
    CompletableFuture<Boolean> AddSnapshot(Snapshot snapshot);
}

File: modules/domain/src/main/kotlin/org/openmedstack/eventstore/IAccessSnapshots.kt
package org.openmedstack.eventstore

import org.openmedstack.domain.Memento
import java.util.concurrent.CompletableFuture

interface IAccessSnapshots {
    fun <TMemento : Memento> getSnapshot(type: Class<Snapshot<TMemento>>, bucketId: String?, streamId: String, maxRevision: Int): CompletableFuture<Snapshot<TMemento>?>
    fun <TMemento : Memento> addSnapshot(snapshot: Snapshot<TMemento>): CompletableFuture<Boolean>
}
